src/main/java/com/ncc/aif/ValidateAIFCli.java

```diff
@@ -0,0 +1,597 @@
+    static final String ERR_TOO_MANY_ONT_FLAGS = "Can only use one of these flags: --ldc | --program | --ont";
+    static final String ERR_MISSING_FILE_FLAG = "Must use one of these flags: -f | -d";
+    static final String ERR_TOO_MANY_FILE_FLAGS = "Can only use one of these flags: -f | -d";
+    static final String ERR_SMALLER_THAN_MIN = "%s must be at least %d";
+    // Logging strings
+    static final String START_MSG = "AIF Validator";
+
+    //=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
+    // Internal Constants
+    // ----------------------------
+    // Logger
+    private static final Logger logger = (Logger) (org.slf4j.LoggerFactory.getLogger(Logger.ROOT_LOGGER_NAME));
+    // ViolationThreshold
+    private static final String ABORT_PARAMETER_STRING = "Abort parameter";
+    private static final int DEFAULT_MAX_VIOLATIONS = -1;
+    private static final int MINIMUM_MAX_VIOLATIONS = 3;
+    // Profiling
+    private static final int LONG_QUERY_THRESH = 2000;
+    // Threading
+    private static final String THREAD_COUNT_STRING = "Thread count";
+    private static final int MINIMUM_THREAD_COUNT = 1;
+
+    //=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
+    // Command Line Arguments
+    // ----------------------------
+    private int argCounter = 1;
+    //TODO: make ArgGroup when 4.0 is stable
+    @Option(names = "--ldc", description = "Validate against the LDC ontology")
+    private boolean useLDCOntology;
+
+    @Option(names = "--program", description = "Validate against the program ontology")
+    private boolean useProgramOntology;
+
+    @Option(names = "--ont", description = "Validate against the OWL-formatted ontolog(ies) at the specified filename(s)",
+            paramLabel = "FILE", arity = "1..*")
+    private List<File> customOntologies;
+
+    //TODO: make ArgGroup when 4.0 is stable
+    @Option(names = "--nist", description = "Validate against the NIST restrictions")
+    private boolean useNISTRestriction;
+
+    @Option(names = "--nist-ta3", description = "Validate against the NIST hypothesis restrictions (implies --nist)")
+    private boolean useNISTTA3Rescriction;
+
+    @Option(names = "-o", description = "Save validation report model to a file.  KB.ttl would result in KB-report.txt.")
+    private boolean outputToFile;
+
+    @Option(names = "--abort", description = "Abort validation after [num] SHACL violations (num > 2), or 3 violations if [num] is omitted.", paramLabel = "num")
+    private int maxValidationErrors = DEFAULT_MAX_VIOLATIONS;
+
+    @Option(names = "--pm", description = "Enable progress monitor that shows ongoing validation progress")
+    private boolean useProgressMonitor;
+
+    @Option(names = "-p", description = "Enable profiling", hidden = true)
+    private boolean useProfiling;
+
+    @Option(names = "--p2", description = "Enable progressive profiling", hidden = true)
+    private boolean useProgressiveProfiling;
```

Comment:
I tried profiling a multi-threaded validation.  It didn't go well due to memory usage:
```
HAL t=3, 26GB, profiler (-p), ran OOM after about 6 minutes.  Memory usage doubled from 13G to 26GB in about 70s.
HAL t=3, 26GB, profiler (-p2), ran OOM after about 9 minutes.  Similar memory spike (10-26G in about 60s).
```

It's not necessarily a requirement that they co-exist, but it's something potentially to investigate and if nothing else, to document.